sweep/txgenerator.go

```diff
@@ -328,34 +275,17 @@ func getWeightEstimate(inputs []input.Input) ([]input.Input, int64, int, int) {
 	for i := range inputs {
 		inp := inputs[i]
 
-		// For fee estimation purposes, we'll now attempt to obtain an
-		// upper bound on the weight this input will add when fully
-		// populated.
-		size, isNestedP2SH, err := getInputWitnessSizeUpperBound(inp)
+		wt := inp.WitnessType()
+		csvCnt, cltvCnt, err := wt.AddWeightEstimation(&weightEstimate)
```

Comment:
I think these counts are now overwritten every time. Need to make a `var err error` and use `=` instead of `:=`

Reply:
Yes, these variables are overwritten every time. But at the end of the loop they are added to the total counts (variables that are outside of the loop and not shown in the diff).
I renamed the vars inside the loop to make it more clear.

Reply:
Ah, they were indeed almost identically named